controllers/compaction/reconciler.go

```diff
@@ -364,7 +369,7 @@ func getCompactionJobVolumeMounts(etcd *druidv1alpha1.Etcd, logger logr.Logger)
 	return vms
 }
 
-func getCompactionJobVolumes(etcd *druidv1alpha1.Etcd, logger logr.Logger) []v1.Volume {
+func getCompactionJobVolumes(ctx context.Context, etcd *druidv1alpha1.Etcd, cl client.Client, logger logr.Logger) []v1.Volume {
```

Comment:
```suggestion
func getCompactionJobVolumes(ctx context.Context, cl client.Client, logger logr.Logger, etcd *druidv1alpha1.Etcd) []v1.Volume {
```

Standard coding practice is to first pass method-related meta parameters, such as context, clients and loggers, followed by helper objects .

Reply:
Done